### PR TITLE
KTOR-6501 Add client KDocs for config, engine, and ObservableContent

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: ktor-client/ktor-client-core/common/src/io/ktor/client/engine/HttpClientEngine.kt [hunk 1]
```diff
@@ -42,13 +100,26 @@ public interface HttpClientEngine : CoroutineScope, Closeable {
...
     /**
-     * Creates a new [HttpClientCall] specific for this engine, using a request [data].
+     * Executes an HTTP request and produces an HTTP response.
+     *
+     * This function is responsible for converting [HttpRequestData], which
+     * contains all the request details, into [HttpResponseData], which encapsulates
+     * the response information such as headers, status code, and body.
+     *
+     * @param data The [HttpRequestData] representing the request to be executed.
+     * @return A [HttpResponseData] object containing the server's response.
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/engine/HttpClientEngine.kt:110]:
```suggestion
     * @return An [HttpResponseData] object containing the server's response.
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/engine/Utils.kt [hunk 2]
```diff
@@ -13,7 +13,7 @@ import kotlinx.coroutines.*
 import kotlin.coroutines.*
 
 /**
- * Default user agent to use in ktor client.
+ * Default user agent to use in a ktor client.
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/engine/Utils.kt:16]:
```suggestion
 * Default user agent to use in a Ktor client.
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/HttpCallValidator.kt [hunk 3]
```diff
@@ -75,9 +75,9 @@ public typealias CallExceptionHandler = suspend (cause: Throwable) -> Unit
 public typealias CallRequestExceptionHandler = suspend (cause: Throwable, request: HttpRequest) -> Unit
 
 /**
- * Response validator plugin is used for validate response and handle response exceptions.
+ * Response validator plugin is used for validating [HttpClient] response and handle response exceptions.
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/HttpCallValidator.kt:78]:
```suggestion
 * The response validator plugin is used for validating an [HttpClient] response and handling response exceptions.
```

File: ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/HttpCallValidator.kt [hunk 4]
```diff
@@ -75,9 +75,9 @@ public typealias CallExceptionHandler = suspend (cause: Throwable) -> Unit
...
  *
- * See also [Config] for additional details.
+ * See also [HttpCallValidatorConfig] for additional details.
```
Comment (vnikolova) [inline ktor-client/ktor-client-core/common/src/io/ktor/client/plugins/HttpCallValidator.kt:80]:
```suggestion
 * For more details, see [HttpCallValidatorConfig].
```

**Review by vnikolova — APPROVED** [review]
Nice, very informative! I added some comments with small improvements and minor grammar corrections.

File: ktor-client/ktor-client-core/common/test/HttpClientConfigTest.kt [hunk 5]
```diff
@@ -0,0 +1,44 @@
+import io.ktor.client.*
+import io.ktor.client.plugins.api.*
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+/*
+ * Copyright 2014-2024 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
```
Comment (osipxd) [inline ktor-client/ktor-client-core/common/test/HttpClientConfigTest.kt:8]: IDEA adds imports above copyright note if a package is not specified. Could you move copyright to the top?

Comment (e5l) [reply ktor-client/ktor-client-core/common/test/HttpClientConfigTest.kt:8]: sure!

**Pull request merged**